File: src/test/java/org/cloudfoundry/promregator/endpoint/InvalidateCacheEndpointTest.java
package org.cloudfoundry.promregator.endpoint;

import org.cloudfoundry.promregator.JUnitTestUtils;
import org.cloudfoundry.promregator.cfaccessor.CFAccessorCacheClassic;
import org.cloudfoundry.promregator.endpoint.MockedAppInstanceScannerEndpointSpringApplication.MockedCFAccessorCache;
import org.cloudfoundry.promregator.endpoint.MockedAppInstanceScannerEndpointSpringApplication.MockedCachingTargetResolver;
import org.cloudfoundry.promregator.scanner.TargetResolver;
import org.junit.AfterClass;
import org.junit.Assert;
import org.junit.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.test.context.TestPropertySource;
import org.springframework.test.context.junit.jupiter.SpringExtension;

@ExtendWith(SpringExtension.class)
@SpringBootTest(classes = MockedAppInstanceScannerEndpointSpringApplication.class)
@TestPropertySource(locations="default.properties")
public class InvalidateCacheEndpointTest {

	@AfterClass
	public static void cleanupEnvironment() {
		JUnitTestUtils.cleanUpAll();
	}

	
	@Autowired
	private InvalidateCacheEndpoint subject;
	
	@Autowired
	private CFAccessorCacheClassic cfAccessorCache;
	
	@Autowired
	private TargetResolver targetResolver;
	
	@Test
	public void testInvalidateCacheAll() {
		Assert.assertNotNull(subject);
		
		ResponseEntity<String> response = subject.invalidateCache(true, true, true, true);
		
		Assert.assertEquals(HttpStatus.NO_CONTENT, response.getStatusCode());
		
		MockedCFAccessorCache cfa = (MockedCFAccessorCache) this.cfAccessorCache;
		Assert.assertTrue(cfa.isApplicationCache());
		Assert.assertTrue(cfa.isOrgCache());
		Assert.assertTrue(cfa.isSpaceCache());
		
		MockedCachingTargetResolver tr = (MockedCachingTargetResolver) this.targetResolver;
		Assert.assertTrue(tr.isResolverCache());
	}

}

File: src/test/java/org/cloudfoundry/promregator/endpoint/InvalidateCacheEndpointTest.kt
package org.cloudfoundry.promregator.endpoint

import io.mockk.called
import io.mockk.mockk
import io.mockk.verify
import org.assertj.core.api.Assertions.assertThat
import org.cloudfoundry.promregator.cfaccessor.CFAccessorCache
import org.cloudfoundry.promregator.scanner.CachingTargetResolver
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.springframework.http.HttpStatus

//@SpringBootTest(classes = MockedAppInstanceScannerEndpointSpringApplication.class)
class InvalidateCacheEndpointTest {
    private lateinit var subject: InvalidateCacheEndpoint
    private val cfAccessorCache = mockk<CFAccessorCache>(relaxed = true)
    private val targetResolver = mockk<CachingTargetResolver>(relaxed = true)

    @BeforeEach
    fun setup() {
        subject = InvalidateCacheEndpoint(cfAccessorCache, targetResolver)
    }

    @Test
    fun testInvalidateCacheAll() {

        val response = subject.invalidateCache(true, true, true, true)
        assertThat(response.statusCode).isEqualTo(HttpStatus.NO_CONTENT)

        verify { cfAccessorCache.invalidateCacheApplications() }
        verify { cfAccessorCache.invalidateCacheOrg() }
        verify { cfAccessorCache.invalidateCacheSpace() }

        verify { targetResolver.invalidateCache() }
    }

    @Test
    fun testInvalidateCacheNone() {

        val response = subject.invalidateCache(false, false, false, false)
        assertThat(response.statusCode).isEqualTo(HttpStatus.NO_CONTENT)

        verify { cfAccessorCache wasNot called }
        verify { targetResolver wasNot called }
    }

}